src/main/cms/cms_menu_persistent_stats.c

```diff
@@ -0,0 +1,107 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+#include <ctype.h>
+
+#include "platform.h"
+
+#ifdef USE_CMS
+#ifdef USE_PERSISTENT_STATS
+
+#include "cms/cms.h"
+#include "cms/cms_types.h"
+#include "cms/cms_menu_persistent_stats.h"
+
+#include "config/config.h"
+#include "pg/stats.h"
+
+uint32_t stats_total_flights;
+uint32_t stats_total_time_s;
+uint32_t stats_total_dist_m;
+int8_t stats_min_armed_time_s;
+
+static const void *cmsx_PersistentStats_onEnter(displayPort_t *pDisp)
+{
+    UNUSED(pDisp);
+
+    stats_total_flights = statsConfig()->stats_total_flights;
+    stats_total_time_s = statsConfig()->stats_total_time_s;
+    stats_total_dist_m = statsConfig()->stats_total_dist_m;
+    stats_min_armed_time_s = statsConfig()->stats_min_armed_time_s;
+
+    return NULL;
+}
+
+static const void *cmsx_PersistentStats_onExit(displayPort_t *pDisp, const OSD_Entry *self)
+{
+    UNUSED(pDisp);
+    UNUSED(self);
+
+    statsConfigMutable()->stats_total_flights = stats_total_flights;
+    statsConfigMutable()->stats_total_time_s = stats_total_time_s;
+    statsConfigMutable()->stats_total_dist_m = stats_total_dist_m;
+    statsConfigMutable()->stats_min_armed_time_s = stats_min_armed_time_s;
+
+    return NULL;
+}
+
+static const void *cmsx_ResetStats(displayPort_t *pDisplay, const void *ptr)
+{
+    UNUSED(ptr);
+
+    stats_total_flights = 0;
+    stats_total_time_s = 0;
+    stats_total_dist_m = 0;
+
+    displayClearScreen(pDisplay);
+    displayRedraw(pDisplay);
+
+    return NULL;
+}
+
+static const OSD_Entry cmsx_menuPersistentStatsEntries[] =
+{
+    {"-- PERSISTENT STATS --", OME_Label, NULL, NULL, 0},
+    {"FLIGHTS", OME_UINT32, NULL, &(OSD_UINT32_t){ &stats_total_flights, 0, UINT32_MAX, 1}, 0},
+    {"TIME(sec)", OME_UINT32, NULL, &(OSD_UINT32_t){ &stats_total_time_s, 0, UINT32_MAX, 1}, REBOOT_REQUIRED},
```

Comment:
Not sure why these would need a reboot?

Reply:
@mikeller if you change these values/settings and exit without saving, the new values will be saved anyway during the next disarm. So I think it's better to be explicit rather than confuse a user "why they were saved when I exited without saving".

Reply:
1. this does not apply if you set `stats_min_armed_time_s` to 0 to disable stats saving;
2. if `stats_min_armed_time_s` is not 0, then this applies to all CMS settings - this is a drawback of stats recording, but not one that can easily be amended.

The purpose of `REBOOT_REQUIRED` is to enforce a reboot for changes that do not apply until after a reboot, or changes that result in an unsafe configuration without a reboot - for what I can tell this is neither of these.